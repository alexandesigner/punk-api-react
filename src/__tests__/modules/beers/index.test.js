/* eslint-disable no-undef */
import configureStore from 'redux-mock-store' // ES6 modules

const middlewares = []
const mockStore = configureStore(middlewares)

const resetState = () => ({ type: 'RESET_STATE' })

describe('Actions', () => {
  it('should dispatch action reset state', () => {
    // Initialize mockstore with empty state
    const initialState = {}

    // MockStor in initialState
    const store = mockStore(initialState)

    // Dispatch the action
    store.dispatch(resetState())

    // Test if your store dispatched the expected actions
    const actions = store.getActions()
    const expectedPayload = { type: 'RESET_STATE' }
    expect(actions).toEqual([expectedPayload])
    expect(actions).toMatchSnapshot()
  })
})
