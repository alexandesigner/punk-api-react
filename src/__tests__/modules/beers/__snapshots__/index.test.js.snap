// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should dispatch action reset state 1`] = `
Array [
  Object {
    "type": "RESET_STATE",
  },
]
`;
